lib/BackbeatConsumer.js

```diff
@@ -778,6 +779,17 @@ class BackbeatConsumer extends EventEmitter {
         }
         this._circuitBreaker.stop();
         return async.series([
+            next => {
+                // stopping consumption of new messages while
+                // waiting for current ones to be processed
+                this._isShuttingDown = true;
+                // waiting for all current jobs to end before exiting
+                if (this._processingQueue && !this._processingQueue.idle()) {
+                    this._processingQueue.drain = next;
```

Comment:
Should we not _call_ drain?
```suggestion
                    this._processingQueue.drain(next);
```

Reply:
Welp... it seems we have multiple dependencies of different versions of the same library `async`.
In version 2.x of `async`, the drain looks like [this](https://github.com/caolan/async/blob/c6bdaca4f9175c14fc655d3783c6af6a883e6514/lib/queue.js#L86-L89):
```
q.drain = function() {
    console.log('all items have been processed');
};
```
And in 3.x, like [this](https://github.com/caolan/async/blob/72de392b6dffc466419ae7cab419d3b120d2459d/lib/queue.js#L78-L81)
```
q.drain(() => {
    console.log('all done')
})
```
I'm not sure yet how the choice of version is made, but the 2.x works locally and on the CI, and the 3.x implementation works on tests in an actual Artesca lab :/

I pushed the code just to test if it passes in the CI, but i stil have to figure out the dependency issue before merging anything.

Reply:
seems something was wrong with the image i was testing with. The code does indeed only use version 2.x